app/src/hooks/useCollateral.tsx

```diff
@@ -0,0 +1,49 @@
+import { useEffect, useState } from 'react'
+
+import { ConnectedWeb3Context } from './connectedWeb3'
+import { getToken } from '../util/addresses'
+import { ERC20Service } from '../services'
+import { Token } from '../util/types'
+import { getLogger } from '../util/logger'
+
+const logger = getLogger('Hooks::useCollateral')
+
+export const useCollateral = (
+  collateralId: KnownToken | string,
+  context: ConnectedWeb3Context,
+): Maybe<Token> => {
+  const [collateral, setCollateral] = useState<Maybe<Token>>(null)
+
+  useEffect(() => {
+    let isSubscribed = true
+
+    const fetchIsValidErc20 = async () => {
+      let collateralData: Maybe<Token> = null
```

Comment:
I guess you are using this name to avoid shadowing the `collateral` variable from `useState`. We don't have a linter rule forbidding that, and in cases like this I think shadowing is fine. But this is just a PP.